recipes/verilator/all/conanfile.py

```diff
@@ -0,0 +1,157 @@
+            }
+            with tools.vcvars(self.settings):
+                with tools.environment_append(build_env):
+                    yield
+        else:
+            yield
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        self._autotools.libs = []
+        self._autotools.library_paths = []
+        if self.settings.compiler == "Visual Studio":
+            self._autotools.cxx_flags.append("-EHsc")
+            self._autotools.defines.append("YY_NO_UNISTD_H")
+            self._autotools.flags.append("-FS")
+        conf_args = [
+            "--datarootdir={}/bin/share".format(tools.unix_path(self.package_folder)),
+        ]
+        self._autotools.configure(args=conf_args, configure_dir=os.path.join(self.build_folder, self._source_subfolder))
+        return self._autotools
+
+    @property
+    def _make_args(self):
+        args = [
+            "CFG_WITH_DEFENV=false",
+            "SRC_TARGET={}".format("dbg" if self.settings.build_type == "Debug" else "opt"),
+        ]
+        if self.settings.build_type == "Debug":
+            args.append("DEBUG=1")
+        if self.settings.compiler == "Visual Studio":
+            args.append("PROGLINK={}".format(tools.unix_path(os.path.join(self.build_folder, self._source_subfolder, "msvc_link.sh"))))
+        return args
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+        try:
+            os.unlink(os.path.join(self._source_subfolder, "src", "config_build.h"))
+        except FileNotFoundError:
+            pass
+
+        if self.settings.compiler == "Visual Studio":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "Makefile_obj.in"),
+                                  "${LINK}", "${PROGLINK}")
+
+    def build(self):
+        self._patch_sources()
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.make(args=self._make_args)
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install(args=self._make_args)
+
+        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "man"))
+        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "verilator", "examples"))
+
+        os.unlink(os.path.join(self.package_folder, "bin", "share", "verilator", "verilator-config-version.cmake"))
+        os.rename(os.path.join(self.package_folder, "bin", "share", "verilator", "verilator-config.cmake"),
+                  os.path.join(self.package_folder, "bin", "share", "verilator", "verilator-tools.cmake"))
```

Comment:
Is there a macro or something like that in those cmake files?

Reply:
Yes indeed. It is tested in the test_package.
Verilator is a installer package. the script will look for a suitable verilator executable and use it to generate sources.

Reply:
I see, thanks for the explanation